packages/cli/src/cog/cog.job.ts

```diff
@@ -0,0 +1,340 @@
+        const bbox = Wgs84.multiPolygonToBbox(coordinates);
+
+        const sourceFs = FileOperator.create(ctx.sourceLocation);
+
+        let description: string | undefined;
+        const providers: StacProvider[] = [];
+        const links: StacLink[] = [
+            {
+                href: FileOperator.join(ctx.outputLocation.path, 'collection.json'),
+                type: 'application/json',
+                rel: 'self',
+            },
+        ];
+        let sourceStac = {} as StacCollection;
+
+        const interval: [string, string][] = [];
+        try {
+            const sourceCollectionPath = FileOperator.join(ctx.sourceLocation.path, 'collection.json');
+            sourceStac = JSON.parse((await sourceFs.read(sourceCollectionPath)).toString()) as StacCollection;
+            description = sourceStac.description;
+            interval.push(...(sourceStac.extent?.temporal?.interval ?? []));
+            links.push({ href: sourceCollectionPath, rel: 'sourceImagery', type: 'application/json' });
+            if (sourceStac.providers != null) providers.push(...sourceStac.providers);
+        } catch (err) {
+            if (!sourceFs.isNotFoundError(err)) {
+                throw err;
+            }
+        }
+        const keywords = sourceStac.keywords ?? CogStacKeywords.slice();
+        const license = sourceStac.license ?? CogStacLicense;
+        const title = sourceStac.title ?? titleizeImageryName(imageryName);
+
+        if (interval.length == 0) {
+            const years = extractYearRangeFromName(imageryName);
+            if (years[0] != -1) {
+                interval.push(years.map((y) => `${y}-01-01T00:00:00Z`) as [string, string]);
+            }
+        }
+
+        if (ctx.cutline) {
+            links.push({ href: 'cutline.geojson.gz', type: 'application/geo+json+gzip', rel: 'cutline' });
+        }
+
+        links.push({ href: 'covering.geojson', type: 'application/geo+json', rel: 'covering' });
+        links.push({ href: 'source.geojson', type: 'application/geo+json', rel: 'source' });
+
+        for (const { name } of metadata.files) {
+            links.push({ href: name + '.json', type: 'application/json', rel: 'item' });
+        }
+
+        const temporal = interval.length == 0 ? undefined : { interval };
+
+        return new CogJob({
+            id,
+            title,
+            description,
+            // eslint-disable-next-line @typescript-eslint/camelcase
+            stac_version: CogStacVersion,
+            // eslint-disable-next-line @typescript-eslint/camelcase
+            stac_extensions: CogStacExtensions,
+
+            extent: {
+                spatial: {
+                    bbox,
+                },
+                temporal,
+            },
+
+            license,
+            keywords,
+
+            providers,
+
+            properties: {
+                created: nowStr,
```

Comment:
is this the prefered way of storing timestamps in here?

Reply:
Yes; though I have dropped this field